Первая попытка

AntonKuzn opened the pull request.

AntonKuzn committed:
Первая попытка

nikolay-977 commented on .idea/.gitignore:
```diff
@@ -0,0 +1,3 @@
+# Default ignored files
+/shelf/
```
Необходимо исправить: при создании пр-а была указана ветка из репозитория заготовки. Это происходит при создании пр-а по умолчанию. При создании пр-а на это необходимо было обратить внимание(об этом упоминается в инструкции). Чтобы это исправить - необходимо в созданном пул-реквесте нажать кнопку edit и указать ветку main своего репозитория вместо ветки main репозитория заготовки.

nikolay-977 commented on .idea/jarRepositories.xml:
```diff
@@ -0,0 +1,20 @@
+  <component name="RemoteRepositoriesConfiguration">
+    <remote-repository>
+      <option name="id" value="central" />
+      <option name="name" value="Central Repository" />
```
Необходимо исправить: в ПР-е не должно быть лишних файлов (.DS_Store, .idea, .build, .target за исключением файлов отчета). Это локальные файлы. Загружать локальные файлы на удаленный репозиторий не принято: это считается плохой практикой. Они не несут полезной информации, но занимают место.

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,84 @@
+    @Test
+    public void testAddIngredient() {
+        burger.addIngredient(ingredientMock);
+        assertEquals(1, burger.ingredients.size());
```
Необходимо исправить: не следует использовать числа, значение которых не является очевидным (антипатерн магические числа). Лучше сохранить в переменную с наименованием, отражающим суть содержимого.

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,84 @@
+
+    @Test
+    public void testMoveIngredient() {
+        Ingredient ingredientSauce = new Ingredient(IngredientType.SAUCE, "sour cream", 200);
```
Необходимо исправить: Тесты для класса Burger должны быть изолированы от классов Bun и Ingredient. При написании тестов для класса Burger во всех кейсах используй моки (тема: Юнит-тесты > методы mockito) и стабы (тема: Юнит-тесты > что такое стаб) классов Bun и Ingredient, а не реальные объекты.

nikolay-977 reviewed (commented):
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта, которое включает название, технологии в проекте (например, Java 11, JUnit 4.13.2, maven 3.9.0 и т.д.), настройка (если необходима) и как запускать (`mvn clean test` - кавычки позволяют выполнять запуск из файла README.md). Как оформлять README.md можно ознакомиться в статье https://blog.jetbrains.com/idea/2022/08/markdown-support-improvements/

AntonKuzn committed:
Create README.md

AntonKuzn committed:
Delete compiler.xml

AntonKuzn committed:
Delete jarRepositories.xml

AntonKuzn committed:
Delete misc.xml

AntonKuzn committed:
Delete vcs.xml

AntonKuzn committed:
Поправил ошибки.
Написал README, удалил лишние файлы (которые смог), перевёл на другой репозиторий (с исходного на свой), поправил антипатерны, изолировал тестовый класс Burger/

AntonKuzn committed:
Delete .idea directory

AntonKuzn committed:
Delete target/maven-archiver directory

AntonKuzn committed:
Delete target/classes/praktikum directory

AntonKuzn committed:
Delete target/maven-status/maven-compiler-plugin directory

AntonKuzn committed:
Delete untitled-1.0-SNAPSHOT.jar

AntonKuzn committed:
Delete target/test-classes directory
